### PR TITLE
Add field date tests

### PR DESCRIPTION
**Pull request by samelhusseini**
Move field date tests from Blockly.
https://github.com/google/blockly/blob/master/tests/mocha/field_date_test.js

Run tests using mocha, custom script as field-date doesn't using ``blockly-scripts``.
Update build script to use externs instead of the goog.provide hack.

### CODE REVIEW

File: plugins/field-date/package.json [hunk 1]
```diff
@@ -5,7 +5,9 @@
     "scripts": {
         "build": "gulp build",
         "clean": "rm -rf $npm_package_directories_dist",
-        "prepublishOnly": "npm run build"
+        "prepublishOnly": "npm run build",
+        "test": "npm run build && npm run test:mocha",
+        "test:mocha": "mocha --ui tdd test/test.mocha.js"
```
Comment (moniika) [inline plugins/field-date/package.json]: Should this be using the blockly-scripts?

File: plugins/field-date/test/test.mocha.js [hunk 2]
```diff
@@ -0,0 +1,159 @@
+/**
```
Comment (moniika) [inline plugins/field-date/test/test.mocha.js]: Other tests in samples like workspace search and slider are using the naming convention for test to be:
field_date_test.mocha.js
I'm not sure if just test or the longer name is better, could update the other ones. I'd just like to be consistent

File: plugins/field-date/test/test.mocha.js [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+ * @license
+ * Copyright 2019 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+const assert = require('assert');
+require('blockly');
```
Comment (moniika) [inline plugins/field-date/test/test.mocha.js]: Why is this needed? and if it is, then maybe field_slider should have it too?

File: plugins/field-date/package.json (hunk 1 above)
Comment (samelhusseini) [reply plugins/field-date/package.json:10]: Oh yes.

Comment (samelhusseini) [reply plugins/field-date/package.json]: No, as blockly-scripts is very much catered for the standard build environment. Field date uses closure to compile not webpack so it's unique.

File: plugins/field-date/test/test.mocha.js (hunk 2 above)
Comment (samelhusseini) [reply plugins/field-date/test/test.mocha.js]: Sure, happy to switch.

File: plugins/field-date/test/test.mocha.js (hunk 3 above)
Comment (samelhusseini) [reply plugins/field-date/test/test.mocha.js]: Not needed, removed.
